catalog/app/components/FileEditor/QuiltConfigEditor/QuiltSummarize/State.tsx

```diff
@@ -0,0 +1,196 @@
+import * as React from 'react'
+
+import quiltSummarizeSchema from 'schemas/quilt_summarize.json'
+
+import type * as Summarize from 'components/Preview/loaders/summarize'
+import { makeSchemaValidator } from 'utils/JSONSchema'
+
+export { default as schema } from 'schemas/quilt_summarize.json'
+
+export interface FileExtended extends Omit<Summarize.FileExtended, 'types'> {
+  isExtended: boolean
+  type?: Summarize.TypeExtended
+}
+
+export interface Column {
+  id: string
+  file: FileExtended
+}
+
+export interface Row {
+  id: string
+  columns: Column[]
+}
+
+export interface Layout {
+  rows: Row[]
+}
+
+const pathToFile = (path: string): FileExtended => ({ path, isExtended: false })
+
+export const emptyFile: FileExtended = pathToFile('')
+
+const createColumn = (file: FileExtended): Column => ({
+  id: crypto.randomUUID(),
+  file,
+})
+
+const createRow = (file: FileExtended): Row => ({
+  id: crypto.randomUUID(),
+  columns: [createColumn(file)],
+})
+
+export const init = (payload?: Layout) => (): Layout =>
+  payload || {
+    rows: [createRow(emptyFile)],
+  }
+
+function insert<T>(array: T[], index: number, item: T): T[] {
```

Comment:
https://effect-ts.github.io/effect/effect/Array.ts.html#insertat

Reply:
And Ramda has `insert` too https://ramdajs.com/docs/#insert

Reply:
the plan was to gradually remove ramda

Reply:
Ok, I didn't have such plans, however I'm happy to not use Ramda, since native JavaScript has almost everything I need for arrays manipulation.